### PR TITLE
Improved definition of Tech Vision

### PR DESCRIPTION
**Pull request by buskamuza**
## This PR is a:

- [ ] New topic
- [x] Content update
- [ ] Content fix or rewrite
- [ ] Bug fix or improvement

## Summary

Added target audience for Tech Vision in the definition.
Eliminated duplication of content between 2.0 and 2.1 versions.

## Additional information

https://devdocs.magento.com/guides/v2.3/coding-standards/technical-vision/index.html

### CODE REVIEW

**Comment by magento-cicd2** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.0/coding-standards/technical-vision/index.md [hunk 1]
```diff
@@ -5,4 +5,9 @@ title: Magento technical vision
...
 
-Each individual technical vision document relates to a set of modules that arelogically grouped together. Individual documents typically describe a component's place in the system, its architecture, extension scenarios. and a list of invariants that must be preserved at all times during component refactoring or extension to ensure consistency.
+Each individual technical vision document relates to a set of modules that are logically grouped together. Individual documents typically describe a component's place in the system, its architecture, extension scenarios. and a list of invariants that must be preserved at all times during component refactoring or extension to ensure consistency.
+
+Technical vision documents are targeting architects and developers to guide them in making decisions based on high-level picture of the system in its desired state.
+
+In contrast, specifications are documents that describe current implementation of the system.
+Though they are also targeting architects and developers, they help with initial investigation, but may be insufficient to make decisions that improve the system.
```
Comment (melnikovi) [inline guides/v2.0/coding-standards/technical-vision/index.md]: Would it make sense to create a list of things technical vision should contain? For instance, architectural principles that current design follows, principles we want to use in the future (desired state), best practices, extension points. Anything else?

Comment (buskamuza) [reply guides/v2.0/coding-standards/technical-vision/index.md]: Technical vision should cover desired state.
It probably makes sense to introduce a template or list some things that are common for a technical vision, I agree. I propose to merge this update as is. I created a separate issue to cover this question https://github.com/magento/architecture/issues/82

**Comment by buskamuza** [comment]
1. Difference between Tech Vision and Tech Guidelines
2. How to group modules?

**Comment by jcalcaben** [comment]
running tests

**Comment by jcalcaben** [comment]
running tests

**Comment by jeff-matthews** [comment]
@jcalcaben, 

We should avoid symlinking to 2.0 content in the future. 

Cc: @dshevtsov

**Pull request merged**